core/controllers/profile.py

```diff
@@ -281,12 +281,17 @@ def post(self):
         """Handles POST requests."""
         username = self.payload.get('username')
         agreed_to_terms = self.payload.get('agreed_to_terms')
+        default_dashboard = self.payload.get('default_dashboard')
         can_receive_email_updates = self.payload.get(
             'can_receive_email_updates')
 
         has_ever_registered = user_services.has_ever_registered(self.user_id)
         has_fully_registered = user_services.has_fully_registered(self.user_id)
 
+        # Set the default dashboard for new users.
```

Comment:
No, this should only happen if the user has _never_ registered in the past, and registration succeeds, right? So basically it should go around or after line 321:

    if not has_ever_registered:
        # do the default dashboard stuff

Note that this handler can get called again by existing users if we update the terms and ask them to agree to them. Or if they click the Back button.